### DISCUSSION TITLE
About the principle of operation

### DISCUSSION THREAD

**Discussion by AiXanadu** [sparkle-project/Sparkle#2307] [Q&A]
Hello everyone, I would like to ask, can someone describe the rationale for the update procedure?
I don't know much about OC.
Is it possible to achieve the same functionality in C++?

Under what kind of operation is this open source project, the application can be updated without entering a password?

**Comment by zorgiepoo** [comment]
> Hello everyone, I would like to ask, can someone describe the rationale for the update procedure?

Sparkle is a software update framework developers use to update their apps.. This is not a good question.

> I don't know much about OC.

I don't know what OC stands for.

> Is it possible to achieve the same functionality in C++?

Sparkle, being written for macOS, uses Apple's APIs which are written in Obj-C. It's feasible to use Sparkle from Obj-C++ (for example we have documentation for using Sparkle from Qt on the website).

> Under what kind of operation is this open source project..

It's open source..

> the application can be updated without entering a password?

Depends. If you ship an app (not a pkg) then for most users yes (unless user is a non-admin user and the app is owned by admin user, or app is owned by root user because some other user feels like doing that).
